drivers/storage/portworx/component/prometheus.go

```diff
@@ -264,6 +265,12 @@ func (c *prometheus) createOperatorClusterRole() error {
 					ResourceNames: []string{"anyuid"},
 					Verbs:         []string{"use"},
 				},
+				{
+					APIGroups:     []string{"policy"},
+					Resources:     []string{"podsecuritypolicies"},
+					ResourceNames: []string{constants.RestrictedPSPName},
+					Verbs:         []string{"use"},
+				},
```

Comment:
Don't we need the PSP for prometheus's cluster role created below on line 279? Did you try to deploy prometheus and check if it comes up without PSPs?

Reply:
Haven't enabled the prometheus operator, going to check this. Also if PodSecurityPolicy plugin is enabled each pod should have an assigned psp to be accepted by kube-apiserver.

Reply:
Prometheus needs some psp too:
```
2s          Warning   FailedCreate                        replicaset/px-prometheus-operator-655df66dfd                     Error creating: pods "px-prometheus-operator-655df66dfd-" is forbidden: unable to validate against any pod security policy: []
```

Reply:
If we need it, then don't we need to add it to the prometheus role as well?